core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -334,6 +334,17 @@ class SparkHadoopUtil extends Logging {
    * Stop the thread that does the delegation token updates.
    */
   private[spark] def stopExecutorDelegationTokenRenewer() {}
+
+  /**
+   * Disable the hadoop fs cache mechanism, otherwise DFSClient will use old token to connect nn.
```

Comment:
also this java doc doesn't really describe what this method does. You need something like:
```
Return a fresh Hadoop configuration, bypassing the HDFS cache mechanism.
This is to prevent the DFSClient from using an old cached token to connect to the NameNode.
```